app/src/main/java/com/github/wnder/picture/LocalPictureDatabase.java

```diff
@@ -0,0 +1,270 @@
+    }
+
+    /**
+     * Update the scoreboard of the picture in the internal storage
+     * @param scoreboard updated scoreboard
+     */
+    public void updateScoreboard(String uniqueId, Map<String, Double> scoreboard){
+        try {
+            //Get and remove old scoreboard
+            String oldMetadata = openMetadataFile(uniqueId);
+            JSONObject json = LocalPictureSerializer.deserializePicture(oldMetadata);
+            json.remove("scoreboard");
+
+            //put in new scoreboard
+            json.put("scoreboard", scoreboard);
+            storeMetadataFile(json.toString(), uniqueId);
+        } catch (JSONException e){
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Tool method used to get real or guessed location using serialization
+     * @param uniqueId uniqueId of picture
+     * @param realOrGuess 0 for real, 1 for guess
+     * @return real or guessed location, depending on realOrGuess
+     */
+    private Location getRealOrGuessed(String uniqueId, int realOrGuess){
+        //get metadata
+        String serializedData = openMetadataFile(uniqueId);
+        JSONObject json = LocalPictureSerializer.deserializePicture(serializedData);
+
+        //get real pic
+        if (realOrGuess == 0) {
+            return LocalPictureSerializer.getRealLocation(json);
+        }
+        //get guess pic
+        else {
+            return LocalPictureSerializer.getGuessLocation(json);
+        }
+    }
+
+    /**
+     * Get the location of the image from the local database
+     * @param uniqueId id of the image
+     * @return the actual location of the image
+     */
+    public Location getLocation(String uniqueId) {
+        return getRealOrGuessed(uniqueId, 0);
+    }
+
+    /**
+     * Get the location of the image from the local database
+     * @param uniqueId id of the image
+     * @return the actual location of the image
+     */
+    public Location getGuessedLocation(String uniqueId) {
+        return getRealOrGuessed(uniqueId, 1);
+    }
+
+    /**
+     * Get the scoreboard of the image from the local database
+     * @param uniqueId id of the image
+     * @return a map of the scoreboard
+     */
+    public Map<String, Double> getScoreboard(String uniqueId) {
+        //get metadata
+        String serializedData = openMetadataFile(uniqueId);
+        try {
+            //deserialize and go back to hashmap
+            JSONObject json = LocalPictureSerializer.deserializePicture(serializedData);
+            String scoreboard = json.getString("scoreboard");
+            return new Gson().fromJson(scoreboard, Map.class);
+        } catch (JSONException e){
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    /**
+     * Reads the metadata file
+     * @param filename uniqueId of picture
+     * @return content of file, empty if there is a problem
+     */
+    private String openMetadataFile(String filename){
```

Comment:
I think keeping "uniqueId" instead of "filename"  would make it more clear/coherent. (Once again, personal opinion)

Reply:
agreed and changed